src/main/java/nl/tudelft/ewi/devhub/server/database/entities/notifications/Notification.java

```diff
@@ -0,0 +1,51 @@
+package nl.tudelft.ewi.devhub.server.database.entities.notifications;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import nl.tudelft.ewi.devhub.server.database.entities.User;
+import org.hibernate.annotations.CreationTimestamp;
+import org.hibernate.validator.constraints.NotEmpty;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+import java.net.URI;
+import java.util.Date;
+
+/**
+ * Created by Arjan on 7-6-2017.
+ */
+@Data
+@Entity
+@Table(name = "notification")
+@EqualsAndHashCode(of = {"id"})
+public class Notification {
+
+    @Id
+    @Column(name = "id")
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private long id;
+
+    @Column(name = "event")
+    @NotNull
+    //Change this to Enum?
+    private String event;
+
+    @Column(name = "message")
+    @NotEmpty
+    private String message;
+
+    @Column(name = "link")
+    // Change this to URI later
+    private String link;
```

Comment:
Actually, change this to `URI` now and see what happens. We should discuss too whether we want to store the link with the notification, or refer to an viewable object instead. Not only is the domain entity now very aware on how its propagated through the DevHub UI, but this approach also fills the database with URI's that may not always remain accessible (if we have to rewrite our API for example). It is perhaps easier to reference to the `Issue`, `Commit`, `Comment` or whatever entity triggered the event and then use the URI inherited from that.

This could be implemented in various ways but its probably nicest to create an inheritance tree of `Notifications` such as `IssueNotification`, `CommitNotification`, etc. This is however not trivial to map in Hibernate (I would have to help with that). I am primarily trying to start a discussion here whether its the best idea to model it this way.